src/game/savegame/savegame.c

```diff
@@ -432,8 +433,13 @@ bool Savegame_Save(int32_t slot_num, GAME_INFO *game_info)
     if (ret) {
         REQUEST_INFO *req = &g_SavegameRequester;
         req->item_flags[slot_num] &= ~RIF_BLOCKED;
-        sprintf(
-            &req->item_texts[req->item_text_len * slot_num], "%s %d",
+        size_t out_size =
+            snprintf(
+                NULL, 0, "%s %d", g_GameFlow.levels[g_CurrentLevel].level_title,
+                g_SaveCounter)
+            + 1;
+        snprintf(
+            req->item_texts[slot_num], out_size, "%s %d",
```

Comment:
Pointless use of `snprintf` – first we calculate how big the string is, then print to the buffer this many bytes, which is the same as just doing `sprintf` directly. It'd be more appropriate if we used the calculated string size to allocate the buffer. However, in this case, we have a preallocated buffer of `req->item_text_len` size. Therefore, we should do `snprintf(req->item_texts[slot_num], req->item_text_len, …)`. Or better yet, postpone the string allocation from `Requester_Init` until we actually need to `snprintf` the strings at call sites, and instead of ceding this responsibility to the callers, wrap the entire operation (calculate item text size → allocate the item text buffer → printf the desired string to the item text buffer) in a new function.